sql-composer/src/types.rs

```diff
@@ -369,6 +410,199 @@ impl fmt::Display for SqlComposition {
+    type Error = Error;
+    fn try_from(path: &Path) -> Result<Self> {
+        SqlComposition::from_path(path)
+    }
+}
+
+/// Implements `TryInto` Trait to convert a PathBuf into a ParsedItemSql.
+///
+/// See TryInfo<Path> implementation for details and motivation.
+///
+/// # Examples
+///
+/// TryInto:
+/// ```
+/// use std::convert::TryInto;
+/// use std::path::PathBuf;
+/// use sql_composer::{types::ParsedItemSql,
+///                    error::Result};
+/// fn main() -> Result<()> {
+///   let path = PathBuf::from("src/tests/simple-template.tql");
+///   let stmt: ParsedItemSql = path.try_into()?;
+///   Ok(())
+/// }
+/// ```
+///
+///
+/// TryFrom:
+/// ```
+/// use std::convert::TryFrom;
+/// use std::path::PathBuf;
+/// use sql_composer::{types::ParsedItemSql,
+///                    error::Result};
+/// fn main() -> Result<()> {
+///   let path = PathBuf::from("src/tests/simple-template.tql");
+///   let stmt = ParsedItemSql::try_from(path)?;
+///   Ok(())
+/// }
+/// ```
+impl TryFrom<PathBuf> for ParsedItemSql {
+    type Error = Error;
+    fn try_from(path: PathBuf) -> Result<Self> {
+        SqlComposition::from_path(path)
+    }
+}
+
+/// Implements `TryInto` Trait to convert a str into a ParsedItemSql.
+///
+/// See TryInfo<Path> implementation for details and motivation.
+///
+/// # Examples
+///
+/// TryInto:
+/// ```
+/// use std::convert::TryInto;
+/// use sql_composer::{types::ParsedItemSql,
+///                    error::Result};
+/// fn main() -> Result<()> {
+///   let path = "src/tests/simple-template.tql";
+///   let stmt: ParsedItemSql = path.try_into()?;
+///   Ok(())
+/// }
+/// ```
+///
+///
+/// TryFrom:
+/// ```
+/// use std::convert::TryFrom;
+/// use sql_composer::{types::ParsedItemSql,
+///                    error::Result};
+/// fn main() -> Result<()> {
+///   let path = "src/tests/simple-template.tql";
+///   let stmt = ParsedItemSql::try_from(path)?;
+///   Ok(())
+/// }
+/// ```
+impl TryFrom<&str> for ParsedItemSql {
+    type Error = Error;
+    fn try_from(path: &str) -> Result<Self> {
+        SqlComposition::from_path(path)
+    }
+}
+
+/// Implements `TryInto` Trait to convert a String into a ParsedItemSql.
+///
+/// See TryInfo<Path> implementation for details and motivation.
+///
+/// # Examples
+///
+/// TryInto:
+/// ```
+/// use std::convert::TryInto;
+/// use sql_composer::{types::ParsedItemSql,
+///                    error::Result};
+/// fn main() -> Result<()> {
+///   let path = String::from("src/tests/simple-template.tql");
+///   let stmt: ParsedItemSql = path.try_into()?;
+///   Ok(())
+/// }
+/// ```
+///
+///
+/// TryFrom:
+/// ```
+/// use std::convert::TryFrom;
+/// use sql_composer::{types::ParsedItemSql,
+///                    error::Result};
+/// fn main() -> Result<()> {
+///   let path = "src/tests/simple-template.tql".to_string();
+///   let stmt = ParsedItemSql::try_from(path)?;
+///   Ok(())
+/// }
+/// ```
+impl TryFrom<String> for ParsedItemSql {
```

Comment:
I feel like `TryFrom<String>` and `TryFrom<&str>` might be a better fit calling parse() from your new interface. Maybe something that supported a `TryInto<PathBuf> ` could be added to keep it simple to parse by passing a filename.